wled00/FX_fcn.cpp

```diff
@@ -437,7 +440,21 @@ uint32_t IRAM_ATTR_YN Segment::currentColor(uint8_t slot) const {
 #endif
 }
 
-void Segment::setCurrentPalette() {
+// pre-calculate drawing parameters for faster access
```

Comment:
@blazoncek I have the feeling you took an idea that I implemented previously in the MM fork - pre-calculate drawing parameters for faster access. It is indeed a big speedup (my test in MM were up to 10% faster).

I don't want to go pricky about this, but it would be nice if you state where you took inspiration from. Something like "based on an idea from @softhack007 in the MM fork". Please.

Reply:
@blazoncek thanks

Reply:
> I don't want to go pricky about this, but it would be nice if you state where you took inspiration from.

None of this would be needed if you'd provide similar optimisations upstream or not adopt GPL in MM. So I had to come up with something similar. You have the liberty (due to MIT) to pick anything from upstream while I need to beg people or figure out something.
Just my 2¢.

Reply:
GPLv3 is not the problem. The problem is MIT licensing in the AC repo, which has become a real no-go for me (and a few others) after some very frustrating experiences. I an still the autor of the code, so I may "provide" it to a repo licensed with MIT, but you're not allowed to "take" without explicit permission. I know this is not optimal and we must find a better approach soon.


@Aircoookie You have proposed a while ago to harmonize the licensing between both repo's, to remove this "rift", by using EUPL-1-2. I'm not sure if there was further discussion on discord. 
The MM team is willing to go this way. If you also agree to change to EUPL-1.2, we (MM team) will start collecting the necessary permissions from contributor since our change to GPLv3. Please give me a sign :-)

Reply:
> The problem is MIT licensing in the AC repo, which has become a real no-go for me (and a few others) _**after some very frustrating experiences**_.

NOTE: Emphasis mine.

Can you explain? Can others mentioned explain as well? I've added about 48k lines of code to WLED (not counting 0_15 branch) but have yet to encounter "frustrating experience". What does "frustrating experience" even mean?
Is that a stolen or "borrowed" code? It inevitably happens with GPL as well.

I am not against license change, I just want to know/understand true reasons for it. Is MIT too permissive? Everyone knew/knows WLED uses MIT so there should be no "frustrating experiences" as a consequence of that IMO. I would really love to hear everyone's perspective, not for WLED project but my personal insights.

Reply:
I can understand you wanting to know more, but I'm not sure that a public discussion as a side topic on a PR is the appropriate place. 

Can we arrange a call/meeting with relevant parties?